lib/puppet/util/windows/adsi.rb

```diff
@@ -201,21 +233,50 @@ def remove_from_groups(*group_names)
     end
     alias remove_from_group remove_from_groups
 
+
+    def add_group_sids(*sids)
+      group_names = []
+      sids.each do |sid|
+        group_names << Puppet::Util::Windows::SID.sid_to_name(sid)
+      end
+
+      add_to_groups(*group_names)
+    end
```

Comment:
Introduced instead of changing up `add_to_groups`. With the aliases I went the safe route but don't necessarily need to.